memgpt/local_llm/vllm/api.py

```diff
@@ -0,0 +1,51 @@
+import os
+from urllib.parse import urljoin
+import requests
+
+from ..utils import load_grammar_file, count_tokens
+
+WEBUI_API_SUFFIX = "/generate"
+DEBUG = False
+
+
+def get_vllm_completion(endpoint, model, prompt, context_window, settings={}, grammar=None):
+    """https://github.com/vllm-project/vllm/blob/main/examples/api_client.py"""
+    prompt_tokens = count_tokens(prompt)
+    if prompt_tokens > context_window:
+        raise Exception(f"Request exceeds maximum context length ({prompt_tokens} > {context_window} tokens)")
+
+    # Settings for the generation, includes the prompt + stop tokens, max length, etc
+    request = settings
+    request["prompt"] = prompt
+    request["max_tokens"] = int(context_window - prompt_tokens)
+    request["stream"] = False
+    # TODO pass model?
```

Comment:
Not sure if the POST endpoint allows model as a parameter? Didn't see it in the documentation so probably requires reading the web server code

Reply:
@cpacker The model is loaded when you start the API server with the `--model` flag